src/connector/src/source/kafka/source/reader.rs

```diff
@@ -106,11 +107,19 @@ impl SplitReader for KafkaSplitReader {
                 .expect("bytes.per.second expect usize"),
         };
 
+        let max_num_messages = match properties.max_num_messages {
+            None => usize::MAX,
+            Some(number) => number
+                .parse::<usize>()
+                .expect("max.num.messages expect usize"),
```

Comment:
We should not panic here since the input is provided by user.

Reply:
it will be used by developers internally to control the number of messages to consume from kafka

Reply:
So the source reader needs to keep a counter for it? When it reaches the number of messages, the reader no longer polls kafka?

Reply:
yes, so that when running benchmarks, we first generate events, say `X` messages that are enough for running queries 30min+. But some query is significantly slower than the rest. So they consume from the same kafka topic but only want to consume `X/50` messages.

The reason we need an end is that the performance dashboard calculates avg-XX metrics based on the e2e time.